### DISCUSSION TITLE
Using source files in tests directory of modules

### DISCUSSION THREAD

**Discussion by ttruster** [idaholab/moose#22642] [Q&A Modules: General] [answered]
I am working on some module and framework development, and I would like to put the source files in the tests directory rather than the main framework as they are specialized rather than general. In particular, I'm in the Tensor Mechanics Module.

I have registered the files in `TensorMechanicsTestApp` like the other files in there. However, when I use either the moose/modules/tensor_mechanics/tensor_mechanics-opt executable or the combined-opt executable, both times I get an error message from running input files:
*** ERROR ***
A 'MyTestObject' is not a registered object.

The same thing happens when I run the modules/tensor_mechanics/test/tests/domain_integral_thermal/interaction_integral_2d_eig_grad.i input file:
*** ERROR ***
A 'FunctionIsotropicEigenstrain' is not a registered object.

The Makefile of those codes has the `USE_TEST_LIBS     := yes` flag. But is there something else I need in order to actually use those compiled objects? For the framework, I know that the moose_test-opt executable does contain the test objects.

**Comment by GiudGiud** [comment, marked as answer]
You ll need to pass --allow-test-object on the command line

same in the tests configuration files iirc

**Reply by GiudGiud** [reply]
actually no it s the other way, there is a flag to disallow them
  --disallow-test-objects                           Don't register test objects and syntax
